tests/periph_pwm/main.c

```diff
@@ -20,42 +20,141 @@
  * every 1s on every channel.
  *
  * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Semjon Kerner <semjon.kerner@fu-berlin.de>
  *
  * @}
  */
 
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
 
 #include "xtimer.h"
+#include "irq.h"
+#include "shell.h"
 #include "timex.h"
 #include "periph/pwm.h"
 
-#define INTERVAL    (10LU * US_PER_MS) /* 10 ms */
-#define STEP        (10)
+#define OSC_INTERVAL    (10LU * US_PER_MS) /* 10 ms */
+#define OSC_STEP        (10)
+#define OSC_MODE        PWM_LEFT
+#define OSC_FREQU       (1000U)
+#define OSC_STEPS       (1000U)
+#define PWR_SLEEP       (1U)
 
-#define MODE        PWM_LEFT
-#define FREQU       (1000U)
-#define STEPS       (1000U)
+bool initiated;
 
+int init(int argc, char** argv) {
+    initiated = false;
 
-int main(void)
-{
+    if (argc != 5) {
+        printf("usage: %s <dev> <mode> <frequency> <resolution>\n", argv[0]);
+        printf("\tdev: device by number between 0 and %d\n", PWM_NUMOF);
+        printf("\tmode:\n");
+        printf("\t\t0: left aligned\n");
+        printf("\t\t1: right aligned\n");
+        printf("\t\t2: center aligned\n");
+        printf("\tfrequency: desired frequency in Hz\n");
+        printf("\tresolution: number between 2 and %d\n", UINT16_MAX);
+        return 1;
+    }
+    errno = 0;
+    pwm_t dev     = (pwm_t)strtol(argv[1], NULL, 10);
+    uint16_t mode = (uint16_t)strtol(argv[2], NULL, 10);
+    uint32_t freq = (uint32_t)strtol(argv[3], NULL, 10);
+    uint16_t res  = (uint16_t)strtol(argv[4], NULL, 10);
+    if (errno != 0) {
+        printf("[pwm] Error: some input was not a number or out of bounds.\n");
+        return 1;
+    }
+
+    if (dev >= PWM_NUMOF) {
+        printf("[pwm] Error: device %d is unknown.\n", dev);
+        return 1;
+    }
+
+    pwm_mode_t pwm_mode;
+    switch(mode) {
+        case(0):
+            pwm_mode = PWM_LEFT;
+            break;
+        case(1):
+            pwm_mode = PWM_RIGHT;
+            break;
+        case(2):
+            pwm_mode = PWM_CENTER;
+            break;
+        default:
+            printf("[pwm] Error: mode %d is not supported.\n", mode);
+            return 1;
+    }
+
+    if (pwm_init(dev, pwm_mode, freq, res) != 0) {
```

Comment:
One more thing I just noticed: I think it would make sense to print the actual set PWM frequency after calling init.